### ISSUE TITLE
Allowed values of optional flags and arguments fail when  `set -u` mode is set

### ISSUE THREAD

**Issue by chovanecadam** [DannyBen/bashly#445]
### Bashly Version

Latest Ruby Gem

### Description

I have a command with optional arguments and flags. Only some values are allowed and none of the flags or arguments are required. However, when no value is passed, the script fails with `unbound variable` error.

The check for the allowed values is done like this:

```bash
  # :command.whitelist_filter
  if [[ -n ${args['ARG']} ]] && [[ ! ${args['ARG']} =~ ^(a|b)$ ]]; then
    printf "%s\n" "ARG must be one of: a, b" >&2
    exit 1
  fi
  if [[ ${args['--option']} ]] && [[ ! ${args['--option']} =~ ^(a|b)$ ]]; then
    printf "%s\n" "--option must be one of: a, b" >&2
    exit 1
  fi
```

This fails with `unbound variable` error if the argument or the flag are not set:

```
$ ./cli test
./cli: line 298: args['ARG']: unbound variable

$ ./cli test a
./cli: line 302: args['--option']: unbound variable
```

The `:command.whitelist_filter` should use ` [[ -n ${args['arg']:-} ]]` instead of  ` [[ -n ${args['arg']} ]]` to check the default values.

### Contents of bashly.yml

```yaml
name: cli
commands:
  - name: test
    args:
      - name: ARG
        allowed:
          - a
          - b

    flags:
      - long: --option
        arg: OPTION
        allowed:
          - a
          - b
```

### Reproduction Steps

```bash
bashly init
bashly add settings

# set `strict: set -u`
sed -i 's/strict: false/strict: set -u/' settings.yml

# copy over bashly.yml

# generate new script
bashly generate

./cli test
```

### Actual Behavior

```
$ ./cli test
./cli: line 298: args['ARG']: unbound variable
```

### Expected Behavior

No error should be shown, because the argument (or flag) are not required.

**Comment by DannyBen** [comment]
Confirmed, thank you for the easily reproducible report.

I [was sure](https://github.com/DannyBen/bashly/blob/708d64e1d963f0cb13abecf48feb1f1156a06379/spec/spec_helper.rb#L33) that all tests (and examples) are tested under the `struct: true` setting, but I see now that examples are not.

My plan is:

- Change `${args['ARG']}` to ` ${args['ARG']:-}` in the `whitelist_filter` view to fix this particular violation
- Ensure that all examples are generated and tested with strict true
- Fix any other failing example in strict mode, which I suspect I will find

BTW - you should use lowercase when naming args (`arg: option` instead of `arg: OPTION`).

**Comment by DannyBen** [comment]
This is fixed now in master branch, and in the edge docker image.

I will hold off on releasing for a day or two, allowing you some time to test it if you can.

**Comment by chovanecadam** [comment]
Thank you for the quick response, I will check it this evening.

**Comment by chovanecadam** [comment]
I have tested the docker and the issue seems to be fixed with args as well as with flags. By the way, I use podman instead of docker, and I had to add `:U` flag to the volume settings to allow it read-write access to the volume. The alias then looked like this:

```bash
alias bashly='docker run --rm -it --user $(id -u):$(id -g) --volume "$PWD:/app:U" docker.io/dannyben/bashly'
```

**Comment by DannyBen** [comment]
Alright, thanks for confirming.
I will release tomorrow.

**Comment by DannyBen** [comment]
This is now fixed in version 1.1.2.
Thanks for reporting.
